_overviews/core/binary-compatibility-of-scala-releases.md

```diff
@@ -7,31 +7,79 @@ partof: binary-compatibility
+So far, that policy is still applicable for Scala 2, although [there exists a proposal to remove the forward compatibility guarantee](https://docs.scala-lang.org/sips/drop-stdlib-forwards-bin-compat.html).
+
+For Scala 3, the minor version is the *second* number in a version, e.g., 2 in v3.2.1.
+The third number is the *patch* version.
+The major version is always 3.
+
+Scala 3 guarantees both backward and forward compatibility across *patch* releases within a single minor release.
+In particular, this applies within an entire LTS series (which will start with Scala 3.3.x).
+
+It also guarantees *backward* compatibility across *minor* releases in the entire 3.x series, but not forward compatibility.
+This means that libraries compiled with any Scala 3.x version can be used in projects compiled with any Scala 3.y version with y >= x.
+
+In addition, Scala 3.x provides backward binary compatibility with respect to Scala 2.13.y.
+Libraries compiled with Scala 2.13.y can be used in projects using Scala 3.x.
+This policy does not apply to experimental Scala 2 features, which notably includes *macros*.
 
-#### Meta
-Note that so far we've only talked about the jars generated by scalac for the standard library and reflection.
-Our policies do not extend to the meta-issue: ensuring binary compatibility for bytecode generated from identical sources, by different version of scalac? (The same problem exists for compiling on different JDKs.) While we strive to achieve this, it's not something we can test in general. Notable examples where we know meta-binary compatibility is hard to achieve: specialisation and the optimizer.
+In general, none of those guarantees apply to *experimental* features and APIs.
 
-In short, we recommend that library authors use [MiMa](https://github.com/lightbend/migration-manager) to verify compatibility of minor versions before releasing.
-Compiling identical sources with different versions of the scala compiler (or on different JVM versions!) could result in binary incompatible bytecode. This is rare, and we try to avoid it, but we can't guarantee it will never happen.
+#### Checking
+For the Scala library artifacts (`scala-library`, `scala-reflect` and `scala3-library`), these guarantees are mechanically checked with [MiMa](https://github.com/lightbend/migration-manager).
+
+The *policies* above extend to libraries compiled by particular Scala compiler versions.
+Every effort is made to preserve the binary compatibility of artifacts produced by the compiler.
+*However*, that cannot be mechanically checked.
+It is therefore possible, due to bugs or unforeseen consequences, that recompiling a library with a different compiler version affects its binary API.
+We cannot *guarantee* that it will never happen.
+
+We recommend that library authors use [MiMa](https://github.com/lightbend/migration-manager) themselves to verify compatibility of minor versions before releasing.
```

Comment:
```suggestion
For the Scala library artifacts (`scala-library`, `scala-reflect` and `scala3-library`), these guarantees are mechanically checked with [MiMa](https://github.com/lightbend/mima).

The *policies* above extend to libraries compiled by particular Scala compiler versions.
Every effort is made to preserve the binary compatibility of artifacts produced by the compiler.
*However*, that cannot be mechanically checked.
It is therefore possible, due to bugs or unforeseen consequences, that recompiling a library with a different compiler version affects its binary API.
We cannot *guarantee* that it will never happen.

We recommend that library authors use [MiMa](https://github.com/lightbend/mima) themselves to verify compatibility of minor versions before releasing.
```

https://github.com/lightbend/mima/commit/906a81775d0972200291b9b5eff73a4bd86b8792